chapters/descriptor_dynamic_offset.adoc

```diff
@@ -0,0 +1,129 @@
+VkBuffer buffer; // size of 32 bytes
+
+VkDescriptorBufferInfo bufferInfo = {
+    buffer,
+    4,      // offset
+    16      // range
+};
+
+VkWriteDescriptorSet writeInfo = {
+    .dstSet = descriptorSet,
+    .descriptorType = VK_DESCRIPTOR_TYPE_STORAGE_BUFFER_DYNAMIC,
+    .pBufferInfo = bufferInfo
+};
+
+vkUpdateDescriptorSets(
+    1,         // descriptorWriteCount,
+    &writeInfo // pDescriptorWrites,
+);
+
+// No dynamic offset
+vkCmdBindDescriptorSets(
+    1,              // descriptorSetCount,
+    &descriptorSet, // pDescriptorSets,
+    0,              // dynamicOffsetCount
+    NULL            // pDynamicOffsets
+);
+----
+
+Our buffer now currently looks like the following:
+
+image::images/descriptor_dynamic_offset_example_a.png[descriptor_dynamic_offset_example_a.png]
+
+Next, a 8 byte dynamic offset will applied at bind time.
+
+[source,c]
+----
+uint32_t offsets[1] = { 8 };
+vkCmdBindDescriptorSets(
+    1,              // descriptorSetCount,
+    &descriptorSet, // pDescriptorSets,
+    1,              // dynamicOffsetCount
+    offsets         // pDynamicOffsets
+);
+----
+
+Our buffer currently looks like the following:
+
+image::images/descriptor_dynamic_offset_example_b.png[descriptor_dynamic_offset_example_b.png]
+
+== Example with VK_WHOLE_SIZE
+
+This time the `VK_WHOLE_SIZE` value will be used for the range. Everything looks the same as the above example except the `VkDescriptorBufferInfo::range`
+
+[source,c]
+----
+VkDescriptorSet descriptorSet; // allocated
+VkBuffer buffer; // size of 32 bytes
+
+VkDescriptorBufferInfo info = {
+    buffer,
+    4,             // offset
+    VK_WHOLE_SIZE  // range
+};
+
+VkWriteDescriptorSet writeInfo = {
+    .dstSet = descriptorSet,
+    .descriptorType = VK_DESCRIPTOR_TYPE_STORAGE_BUFFER_DYNAMIC,
+    .pBufferInfo = bufferInfo
+};
+
+vkUpdateDescriptorSets(
+    1,         // descriptorWriteCount,
+    &writeInfo // pDescriptorWrites,
+);
+
+// No dynamic offset
+vkCmdBindDescriptorSets(
+    1,              // descriptorSetCount,
+    &descriptorSet, // pDescriptorSets,
+    0,              // dynamicOffsetCount
+    NULL            // pDynamicOffsets
+);
+----
+
+Our buffer currently looks like the following:
+
+image::images/descriptor_dynamic_offset_example_c.png[descriptor_dynamic_offset_example_c.png]
+
+This time, if we attempt to apply a dynamic offset it will be met with undefined behavior and the link:https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/2846[validation layers will give an error]
+
+[source,c]
+----
+// Invalid
+uint32_t offsets[1] = { 8 };
+vkCmdBindDescriptorSets(
+    1,              // descriptorSetCount,
+    &descriptorSet, // pDescriptorSets,
+    1,              // dynamicOffsetCount
+    offsets         // pDynamicOffsets
+);
+----
+
+This is what it looks like with the invalid dynamic offset
+
+image::images/descriptor_dynamic_offset_example_d.png[descriptor_dynamic_offset_example_d.png]
```

Comment:
In the image, the start of the range is offset=4, but seems it should be offset=12 (4 static + 8 dynamic)?

Reply:
good catch, I see how this happen, will fix it asap